### ISSUE TITLE
Fusebox 4.0.0-next.412 regression

### ISSUE THREAD

**Issue by bogdandv** [fuse-box/fuse-box#1862]
When using any version between 4.0.0-next.404 and 4.0.0-next.411 code get built properly.
in version 4.0.0-next.412 one functionality of our application causes an infinite loop.

Please note the only thing that was changed between tests were the fuse version.
Please note that the error happens in react/redux/saga

**Comment by charsleysa** [comment]
@bogdandv does the issue occur with 413?

**Comment by bogdandv** [comment]
Yes, issue does occur in 413

**Comment by charsleysa** [comment]
@bogdandv can you please provide a minimal reproduction?

**Comment by nchanged** [comment]
@bogdandv there has been a good refactor of the compiler API. Please help debug it and I will cover it with tests.

**Comment by bogdandv** [comment]
Honestly I don't know how to replicate it minimally. I can tell you that the issue is in vendor (more exactly the takeEvery function in the saga midleware)
```

615: function(__fusereq, exports, module){
exports.__esModule = true;
var fsmIterator_1 = __fusereq(671);
var fsmIterator_1d = __fuse.dt(fsmIterator_1);
var io_1 = __fusereq(81);
var channel_1 = __fusereq(77);
function takeEvery(patternOrChannel, worker) {
  for (var _len = arguments.length, args = Array(_len > 2 ? _len - 2 : 0), _key = 2; _key < _len; _key++) {
    args[_key - 2] = arguments[_key];
  }
  var yTake = {
    done: false,
    value: io_1.take(patternOrChannel)
  };
  var yFork = function yFork(ac) {
    return {
      done: false,
      value: io_1.fork.apply(undefined, [worker].concat(args, [ac]))
    };
  };
  var action = void 0, setAction = function setAction(ac) {
    return action = ac;
  };
  return fsmIterator_1d.default({
    q1: function q1() {
      return ['q2', yTake, setAction];
    },
    q2: function q2() {
      return action === channel_1.END ? [fsmIterator_1.qEnd] : ['q1', yFork(action)];
    }
  }, 'q1', 'takeEvery(' + fsmIterator_1.safeName(patternOrChannel) + ', ' + worker.name + ')');
}
exports.default = takeEvery;

},
```

worker is undefined

**Comment by nchanged** [comment]
@bogdandv could you paste the source of that file?  The path should above 

```
// path should be here
615: function(__fusereq, exports, module){
```

**Comment by bogdandv** [comment]
// node_modules/redux-saga/es/internal/sagaHelpers/takeEvery.js @615

**Comment by nchanged** [comment]
@bogdandv what's the version you're using? I can't find this source

**Comment by nchanged** [comment]
@bogdandv the output it identical could you maybe reproduce it better?

**Comment by nchanged** [comment]
@bogdandv maybe this could help? https://www.diffchecker.com/diff

if you make a build in both versions and compare the output (but clear the cache first to be sure) The file order might be off in a few places also module numbering but you could spot some issues related to imports.

**Comment by bogdandv** [comment]
my code bundles in 100.000 lines of app and 90.000 lines of vendor code. There are multiple differences regarding export numbering and some examples of bundled depnedencies being created in a different order.

I can not just parse all the differences and find the issue...

**Comment by nchanged** [comment]
@bogdandv there must be a way for us to debug the problem. Please find the reference to that `takeEvery` function. You can do it by simply searching for `takeEvery` in your `vendor` or `app`

I know it's painful, but I believe that only bug from holding us off from making it stable.

**Comment by bogdandv** [comment]
note that both prod and dev modes cause the same errors: callstack exceeded

**Comment by nchanged** [comment]
@bogdandv could you paste the stack trace?

**Comment by bogdandv** [comment]
sagaManager.js:51 TypeError: Cannot read property 'name' of undefined
    at takeEvery (vendor.js:32790)
    at createTaskIterator (vendor.js:32261)
    at runForkEffect (vendor.js:32526)
    at runEffect (vendor.js:32450)
    at vendor.js:32616
    at Array.forEach (<anonymous>)
    at runAllEffect (vendor.js:32615)
    at runEffect (vendor.js:32450)
    at next (vendor.js:32370)
    at proc (vendor.js:32345)
(anonymous) @ sagaManager.js:51
step @ sagaManager.js:24
(anonymous) @ sagaManager.js:5
next @ vendor.js:32354
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
Show 170 more frames
sagaManager.js:51 RangeError: Maximum call stack size exceeded
    at proc (vendor.js:32297)
    at runForkEffect (vendor.js:32533)
    at runEffect (vendor.js:32450)
    at vendor.js:32616
    at Array.forEach (<anonymous>)
    at runAllEffect (vendor.js:32615)
    at runEffect (vendor.js:32450)
    at next (vendor.js:32370)
    at proc (vendor.js:32345)
    at resolveIterator (vendor.js:32466)
(anonymous) @ sagaManager.js:51
step @ sagaManager.js:24
(anonymous) @ sagaManager.js:5
next @ vendor.js:32354
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
task.cont @ vendor.js:32217
next @ vendor.js:32380
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
runEffect @ vendor.js:32450
(anonymous) @ vendor.js:32616
runAllEffect @ vendor.js:32615
runEffect @ vendor.js:32450
next @ vendor.js:32370
proc @ vendor.js:32345
resolveIterator @ vendor.js:32466
runCallEffect @ vendor.js:32506
runEffect @ vendor.js:32450
next @ vendor.js:32370
currCb @ vendor.js:32433
end @ vendor.js:32407
abort @ vendor.js:32206
runForkEffect @ vendor.js:32541
Show 170 more frames
vendor.js:28038 uncaught at abordable at abordable 
 at  
 RangeError: Maximum call stack size exceeded
    at proc (https://localhost:44361/vendor.js:32297:14)
    at resolveIterator (https://localhost:44361/vendor.js:32466:5)
    at runCallEffect (https://localhost:44361/vendor.js:32506:100)
    at runEffect (https://localhost:44361/vendor.js:32450:635)
    at next (https://localhost:44361/vendor.js:32370:9)
    at Object.currCb [as cont] (https://localhost:44361/vendor.js:32433:7)
    at end (https://localhost:44361/vendor.js:32407:23)
    at abort (https://localhost:44361/vendor.js:32206:5)
    at Object.task.cont (https://localhost:44361/vendor.js:32217:9)
    at next (https://localhost:44361/vendor.js:32380:16)

**Comment by nchanged** [comment]
@bogdandv
 is it possible for me to take a look at your vendor.js ?  I don't think there is anything sensitive  related to your project.

**Comment by bogdandv** [comment]
there is an inhouse built project that is included in the vendor and is not open sourced...

**Comment by nchanged** [comment]
Right, then could you check those lines of code

```
    at createTaskIterator (vendor.js:32261)
    at runForkEffect (vendor.js:32526)
    at runEffect (vendor.js:32450)
    at vendor.js:32616
```

if you could send me those 3 file name (where the breakpoint it), it would be great. I will do everything else to compare the outputs.

**Comment by bogdandv** [comment]
// node_modules/redux-saga/es/internal/proc.js @613

**Comment by nchanged** [comment]
I think we need deeper. If you can paste all affected files in the stack

**Comment by nchanged** [comment]
@bogdandv `fuse-box@4.0.0-next.414` There was a severe bug which now has been tested.
Please give it a try.

**Comment by bogdandv** [comment]
Problem solved in fuse-box@4.0.0-next.414
